### ISSUE TITLE
Drop `es/lints` except critical lints

### ISSUE THREAD

**Issue by kdy1** [swc-project/swc#5218]
We cannot replace eslint because of plugins and the conclusion of the discussion was to drop it

**Comment by ArturAralin** [comment]
Maybe I'll make own repo with non critical lints based on swc crate

**Comment by alexander-akait** [comment]
I think time to create `swc_linters` (i.e. `swc_ecma_lint`/`swc_css_lint` and maybe `swc_html_lint` in future), so it will be more scalable and will not distract the main `swc` repo and team (in future).

In long term I think we should replace `eslint`/`stylelint`, our parser and AST are great, perf is better, it will happen sooner or later anyway

**Comment by kdy1** [comment]
I'll not remove rules from the repository, I'll just move it to `@swc/lints`

I'll review your PRs soon

**Comment by Austaras** [comment]
Is swc cureently bundling its own wasm runtime? Does that mean after seperation user need to keep two identical wasm runtime on disk?

**Comment by IWANABETHATGUY** [comment]
> 

Any plan to support **code fix**/**code suggestion** like eslint?

**Comment by ArturAralin** [comment]
@IWANABETHATGUY lints will not have any formatting checks 

https://github.com/swc-project/swc/pull/3888#issuecomment-1061515898

**Comment by IWANABETHATGUY** [comment]
> @IWANABETHATGUY lints will not have any formatting checks
> 
> [#3888 (comment)](https://github.com/swc-project/swc/pull/3888#issuecomment-1061515898)

Nope, **code fix** not equals to  formatting checks.

**Comment by IWANABETHATGUY** [comment]
> @IWANABETHATGUY lints will not have any formatting checks
> 
> [#3888 (comment)](https://github.com/swc-project/swc/pull/3888#issuecomment-1061515898)

But thanks for explaining.

**Comment by IWANABETHATGUY** [comment]
I mean something like this.
![image](https://user-images.githubusercontent.com/17974631/180221990-8505327e-5886-42e6-bcce-a3c2993f19f4.png)

**Comment by Austaras** [comment]
> No we don't bundle wasm

Would #5350 result in final binary?

**Comment by kdy1** [comment]
No, but we may replace native bindings for some platforms with wasm runtime.
Until stabilizing it, we will bundle wasm runtime (`@swc/wasm`) for those platforms.

(Note: We bundle `wasmer`, which is also a wasm runtime. So we may need to distinguish them)

**Comment by kdy1** [comment]
Oh, if you mean `wasmer`, the linter does not use wasm runtime at the moment.

**Comment by Austaras** [comment]
But if we're going to make `@swc/lint` a general purpose linter, plugin system is required. That would make it relies on wasmer too I guess

**Comment by kdy1** [comment]
Yeah, but when we have enough resources for providing general plugins apis for linter, we may move it back to core
Alternatively we can use a shared dylib, though
